### PR TITLE
prover: add `ProcessorTable`

### PR DESCRIPTION
**Pull request by tcoratger**
Should close https://github.com/kkrt-labs/stwo-brainfuck/issues/67

### CODE REVIEW

File: crates/brainfuck_prover/src/components/processor/table.rs [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+/// register values for each cycle that the program ran.
+///
+/// Each row contains the values for the following registers:
+/// - `clk`: The cycle count, which increases by one per step.
+/// - `ip`: The instruction pointer.
+/// - `ci`: The current instruction.
+/// - `ni`: The next instruction.
+/// - `mp`: The memory pointer.
+/// - `mv`: The memory value.
+/// - `mvi`: The memory value increment flag.
```
Comment (zmalatrax) [inline crates/brainfuck_prover/src/components/processor/table.rs:17]:
```suggestion
/// - `mvi`: The memory value inverse.
```

File: crates/brainfuck_prover/src/components/processor/table.rs [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+use stwo_prover::core::fields::m31::BaseField;
+
+/// Represents a single row in the Processor Table.
+///
+/// The Processor Table ensures consistency for the part of the execution that
+/// relates to the registers of the Brainfuck virtual machine. It records all
+/// register values for each cycle that the program ran.
+///
+/// Each row contains the values for the following registers:
+/// - `clk`: The cycle count, which increases by one per step.
```
Comment (zmalatrax) [inline crates/brainfuck_prover/src/components/processor/table.rs:11]:
```suggestion
/// - `clk`: The clock cycle counter, represents the current step.
```

File: crates/brainfuck_prover/src/components/processor/table.rs [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+    ip: BaseField,
+    /// Current instruction: the opcode at the current instruction pointer.
+    ci: BaseField,
+    /// Next instruction: the opcode that follows `ci`, or 0 if out of bounds.
+    ni: BaseField,
+    /// Memory pointer: points to the current cell in the memory array.
+    mp: BaseField,
+    /// Memory value: the value at the current memory cell.
+    mv: BaseField,
+    /// Memory value increment flag: indicates if the memory value should be incremented.
```
Comment (zmalatrax) [inline crates/brainfuck_prover/src/components/processor/table.rs:32]:
```suggestion
    /// Memory value inverse: the modular inverse of `mv` on the cell values' range (over [`BaseField`])
```

File: crates/brainfuck_prover/src/components/processor/table.rs [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+/// - `clk`: The cycle count, which increases by one per step.
+/// - `ip`: The instruction pointer.
+/// - `ci`: The current instruction.
+/// - `ni`: The next instruction.
+/// - `mp`: The memory pointer.
+/// - `mv`: The memory value.
+/// - `mvi`: The memory value increment flag.
+#[derive(Debug, Default, PartialEq, Eq, Clone)]
+pub struct ProcessorTableRow {
+    /// Cycle count: increases by one per step.
```
Comment (zmalatrax) [inline crates/brainfuck_prover/src/components/processor/table.rs:20]:
```suggestion
    /// Clock cycle counter: the current step.
```

File: crates/brainfuck_prover/src/components/processor/table.rs [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use brainfuck_vm::registers::Registers;
+    use num_traits::{One, Zero};
+
+    #[test]
+    fn test_processor_table_row_from_registers() {
+        // Create a Registers instance with test values
```
Comment (zmalatrax) [inline crates/brainfuck_prover/src/components/processor/table.rs:90]: Are comments in these unit tests necessary?
It seems clear enough without them

File: crates/brainfuck_prover/src/components/processor/table.rs [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+            clk: BaseField::one(),
+            ip: BaseField::from(5),
+            ci: BaseField::from(43),
+            ni: BaseField::from(91),
+            mp: BaseField::from(2),
+            mv: BaseField::from(7),
+            mvi: BaseField::zero(),
+        };
+
+        assert_eq!(row, expected_row, "Row should match the expected values");
```
Comment (zmalatrax) [inline crates/brainfuck_prover/src/components/processor/table.rs:115]: Do we need a custom string in the assert_eq! macros when there is only one assertion per test?

We merged a few test with this, but I think we should actually remove them (the custom string)

File: crates/brainfuck_prover/src/components/processor/table.rs (hunk 5 above)
Comment (tcoratger) [reply crates/brainfuck_prover/src/components/processor/table.rs]: yep let us remove them, done

File: crates/brainfuck_prover/src/components/processor/table.rs (hunk 6 above)
Comment (tcoratger) [reply crates/brainfuck_prover/src/components/processor/table.rs]: I just removed them. I think that where they are we can leave them, it logs a message in the console. Sometimes for the CI it is a little easier to find your way around: when you read the error message on the CI among multiple tests failing, it is possible that with the message you know directly what is wrong by remembering exactly the context :)

Comment (zmalatrax) [reply crates/brainfuck_prover/src/components/processor/table.rs]: Okay, makes sense :)

**Review by zmalatrax — APPROVED** [review]

**Pull request merged**